repository.py

```diff
@@ -0,0 +1,543 @@
+#!/usr/bin/env python
+
+# Taken from https://opendev.org/openstack/sunbeam-charms/src/commit/5b37e0a6919668f23b8c7b148717714889fd4381/repository.py
+
+import logging
+import argparse
+import dataclasses
```

Comment:
Why not just `from dataclasses import dataclass`? Looks like we're only using the `dataclass` decorator here.